app/background.ts

```diff
@@ -0,0 +1,90 @@
+/// <reference path="../node_modules/@types/chrome/chrome-app.d.ts" />
+import { RequestObj } from './request_list'
+
+interface Data {
+  tabId: number,
+  enabled: boolean,
+  requests: Array<RequestObj>
+  count: number
+};
+
+const DEFAULT_DATA : Data = {
+  tabId: -1,
+  enabled: false,
+  requests: [],
+  count: 0
+};
+
+class BackgroundWorker {
+  data : Array<Data> = [];
+
+  findItem = (tabId : number) => {
+    for (let index = 0; index < this.data.length; index++) {
+      const data = this.data[index];
+
+      if (Boolean(data) && data.tabId === tabId) {
+        return data;
+      }
+    }
+
+    // Lazy code for object cloning
+    return JSON.parse(JSON.stringify(DEFAULT_DATA));
+  }
+
+  startMessageListener() {
+    chrome.runtime.onMessage.addListener((request, _, sendResponse) => {
+      const tabId = request.tabId;
+      const data = this.findItem(tabId);
```

Comment:
True. We might want this for the other branches too, since a tab id is pretty much a requirement for getting tab related info. Anyway, will remove this.